app/models/concerns/order_status.rb

```diff
@@ -93,6 +92,30 @@ module OrderStatus # rubocop:disable Metrics/ModuleLength
       event :close do
         transition received: :closed
       end
+
+      event :cancel do
+        old_status = ""
+
+        before do
+          old_status = status
+        end
```

Comment:
Rails provides methods for columns to obtain the previous values of columns when saving state. Do those work in the `after` callback? If so, I feel that would be preferable to the `old_status` variable. See an example of it here: https://stackoverflow.com/a/607139/122

Reply:
I tried getting this to work and couldn't. I would much rather do it this way. Here is was it output:
```
[1] pry(#<Order>)> status
=> "canceled"
[2] pry(#<Order>)> status_was
=> "canceled"
[3] pry(#<Order>)> self.status
=> "canceled"
[4] pry(#<Order>)> self.status_was
=> "canceled"
[5] pry(#<Order>)> self.status_changed?
=> false
[6] pry(#<Order>)>```